config/routes.rb

```diff
@@ -21,11 +21,31 @@
 
   get "refresh-session", to: "sessions#refresh", as: :refresh_session
 
-  scope path: ":policy", constraints: {policy: %r{student-loans}} do
-    constraints slug: %r{#{StudentLoans::SlugSequence::SLUGS.join("|")}} do
-      resources :claims, only: [:show, :update], param: :slug, path: "/"
+  # Used to constrain claim journey routing so only slugs
+  # that are part of a policy’s slug sequence are routed.
+  class RestrictToSequenceSlugs
```

Comment:
You can define a class anywhere! The code for it just gets evaluated, just like every other line of Ruby code. And if the place where you've defined it is the only place you use it, then it's probably fine.

The reason we generally put classes in their own files and name those files based on the class name is a) so we have a conventional way of locating the source file, and b) so that the [Rails autoloading magic](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html) knows how to load a class if it hasn't been explicitly `required (by default, Ruby doesn't magically load classes for you; that's entirely a Rails thing).

In this case, this class is very specifically being used here in the routes. It's never going to be called/used outside of this file, so there isn't much to be gained by sticking it in a separate file. Another reason why it might have made sense to put this class in its own file would be if we had unit tests around it, but I opted to rely on the routing specs to verify the behaviour, and because of the higher-level coverage those provide, it didn't feel useful to have lower level unit tests for this class.

Reply:
Excellent! Thanks for explaining!